xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp

```diff
@@ -22,13 +22,35 @@
 
 #include <EGL/egl.h>
 
+#include "cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h"
 #include "utils/log.h"
 
+#if defined(HAVE_LIBVA)
+#include "cores/VideoPlayer/DVDCodecs/Video/VAAPI.h"
+#include "cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGL.h"
+#endif
+
 using namespace KODI::WINDOWING::WAYLAND;
 
 bool CWinSystemWaylandEGLContextGL::InitWindowSystem()
 {
-  return CWinSystemWaylandEGLContext::InitWindowSystemEGL(EGL_OPENGL_BIT, EGL_OPENGL_API);
+  if (!CWinSystemWaylandEGLContext::InitWindowSystemEGL(EGL_OPENGL_BIT, EGL_OPENGL_API))
+  {
+    return false;
+  }
+
+  CLinuxRendererGL::Register();  
+  
+#if defined(HAVE_LIBVA)
+  bool general, hevc;
+  CRendererVAAPI::Register(GetVaDisplay(), m_eglContext.m_eglDisplay, general, hevc);
```

Comment:
> so even if a compositor one day supports a desktop that spans multiple GPUs

I am not talking about this kind of configuration. On X11 you can have multiple displays (or call it desktop) and each one is connected to a different monitor. 
This configuration is not at all hacky on X11 (at least not with nvidia cards I have tested with). You cannot move the window to the other display but you can select the other one in Kodi's settings. Then the app has to switch.